sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLoggerProvider.java

```diff
@@ -48,10 +49,12 @@ public static SdkLoggerProviderBuilder builder() {
       Resource resource,
       Supplier<LogLimits> logLimitsSupplier,
       List<LogRecordProcessor> processors,
-      Clock clock) {
+      Clock clock,
+      ScopeConfigurator<LoggerConfig> loggerConfigurator) {
     LogRecordProcessor logRecordProcessor = LogRecordProcessor.composite(processors);
     this.sharedState =
-        new LoggerSharedState(resource, logLimitsSupplier, logRecordProcessor, clock);
+        new LoggerSharedState(
+            resource, logLimitsSupplier, logRecordProcessor, clock, loggerConfigurator);
```

Comment:
I know this was already the case, and I know that so much of this is just a massive pile of builders and factories, but I _do_ need to point out that this constructor isn't really built with its _true_ dependencies exposed, and is therefore counter to dependency injection (DI) principles.

Can we change the constructor to take a `LoggerSharedState` and `ComponentRegistry` and the noop flag, and then change to builder to build those and pass them in?

Reply:
Sure. To me this is six to one, half dozen the other. But if I go with [my comment here](https://github.com/open-telemetry/opentelemetry-java/pull/6375#discussion_r1567792899), then ScopeConfigurator will be an actual dependency of SdkLoggerProvider, and so this type of refactoring would thus be better in a separate PR.